mlir/test/Conversion/MIGraphXToTosa/mixr-to-tosa-ops.mlir

```diff
@@ -215,11 +222,21 @@ module  {
   // CHECK: tosa.matmul
   // CHECK: tosa.mul
   func.func @func_dot_mul(%arg0: tensor<1x5x4xf32>, %arg1: tensor<1x4x3xf32>, %arg2: tensor<1x5x3xf32>) -> tensor<1x5x3xf32> attributes{kernel, arch = ""} {
-    %0 = "migraphx.dot"(%arg0, %arg1) : (tensor<1x5x4xf32>, tensor<1x4x3xf32>) -> tensor<1x5x3xf32>
+    %0 = migraphx.dot(%arg0, %arg1) : (tensor<1x5x4xf32>, tensor<1x4x3xf32>) -> tensor<1x5x3xf32>
     %2 = "migraphx.mul"(%0, %arg2) {} : (tensor<1x5x3xf32>, tensor<1x5x3xf32>)-> tensor<1x5x3xf32>
     return %2 : tensor<1x5x3xf32>
   }
 
+  // CHECK-LABEL: func.func @func_mul_bcast
+  // CHECK: tosa.reshape{{.*}}1x10x4
+  // CHECK: tosa.reshape{{.*}}1x4x3
+  // CHECK: tosa.matmul
+  // CHECK: tosa.reshape{{.*}}2x5x3
+  func.func @func_mul_bcast(%arg0: tensor<2x5x4xf32>, %arg1: tensor<1x4x3xf32>, %arg2: tensor<2x5x3xf32>) -> tensor<2x5x3xf32> attributes{kernel, arch = ""} {
```

Comment:
Thanks! this should've been here :) .
However, I was more asking for CHECK-NOT: tosa.reshape as you were saying w/o your change a redundant reshape is added but I dont want to hold this on that account.

Reply:
You are right. On a second look the reshape has already been checked over here:

https://github.com/ROCmSoftwarePlatform/rocMLIR/blob/db5e885579222a7cf28b5158bd70d78dbd5dec83/mlir/test/Conversion/MIGraphXToTosa/mixr-to-tosa-ops.mlir#L73-L95

I was wrong to say that this is an unrealistic use case. Since this test is better than the one I have written, I deleted the case in f3ab7eb.

I also added the `CHECK-NOT` according to your suggestion.